lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart

```diff
@@ -21,99 +21,78 @@ class HalvesEventsDisplay extends StatelessWidget {
         color: AppColors.listTileGrey,
       ),
-      height: 200,
-      child: Row(
+      child: ListView(
+        shrinkWrap: true,
+        primary: false,
         children: [
-          Expanded(
-            flex: 1,
-            child: Column(
-              children: [
-                SizedBox(
-                  height: 200,
-                  child: ListView(
-                    physics: const NeverScrollableScrollPhysics(),
-                    children: [
-                      if (isHalfTime == true)
-                        for (final event in matchEvents) ...[
-                          if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
-                            if (homeTeamID == event.team?.id && event.time!.elapsed! < 45)
-                              Padding(
-                                padding: const EdgeInsets.only(left: 8.0, top: 8.0),
-                                child: EventTextTemplate(
-                                    leadingProperty:
-                                        event.time?.elapsed.toString() ?? AppConstVariables.stringPlaceholder,
-                                    player: event.player?.name ?? AppConstVariables.stringPlaceholder,
-                                    isHomeTeam: true),
-                              )
-                            else
-                              const SizedBox.shrink(),
-                        ]
-                      else
-                        for (final event in matchEvents) ...[
-                          if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
-                            if (homeTeamID == event.team?.id && event.time!.elapsed! > 45)
-                              Padding(
-                                padding: const EdgeInsets.only(left: 8.0, top: 8.0),
-                                child: EventTextTemplate(
-                                    leadingProperty:
-                                        event.time?.elapsed.toString() ?? AppConstVariables.stringPlaceholder,
-                                    player: event.player?.name ?? AppConstVariables.stringPlaceholder,
-                                    isHomeTeam: true),
-                              )
-                            else
-                              const SizedBox.shrink(),
-                        ]
-                    ],
-                  ),
-                )
-              ],
-            ),
+          Row(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Expanded(
+                flex: 1,
+                child: Wrap(
+                  direction: Axis.vertical,
+                  children: [
+                    if (isHalfTime == true)
+                      for (final event in matchEvents) ...[
+                        if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
```

Comment:
I think there's too many nesting ifs, maybe you could change it to something like:
`if (opposite of a condition you want) {}`
`    `
`if(second condition you want)`
`   Widgets()`

Instead of
`if (condition you want)`
`  if (second condition you want)`
`     Widgets()`

Reply:
hard to do that because display depends on every condition written here, maybe I could create more simple conditions in the future, but rn I have no idea how to write them in a different way to make them both work and be more readable